contracts/modules/grouping/GroupingModule.sol

```diff
@@ -0,0 +1,227 @@
+            revert Errors.GroupingModule__GroupIPHasNoLicenseTerms(groupIpId);
+        }
+        (address groupLicenseTemplate, uint256 groupLicenseTermsId) = LICENSE_REGISTRY.getAttachedLicenseTerms(
+            groupIpId,
+            0
+        );
+        PILTerms memory groupLicenseTerms = IPILicenseTemplate(groupLicenseTemplate).getLicenseTerms(
+            groupLicenseTermsId
+        );
+        if (groupLicenseTerms.defaultMintingFee != 0) {
+            revert Errors.GroupingModule__GroupIPHasMintingFee(groupIpId, groupLicenseTemplate, groupLicenseTermsId);
+        }
+
+        GROUP_IP_ASSET_REGISTRY.addGroupMember(groupIpId, ipIds);
+        IGroupRewardPool pool = IGroupRewardPool(GROUP_IP_ASSET_REGISTRY.getGroupRewardPool(groupIpId));
+        for (uint256 i = 0; i < ipIds.length; i++) {
+            if (GROUP_IP_ASSET_REGISTRY.isRegisteredGroup(ipIds[i])) {
+                revert Errors.GroupingModule__CannotAddGroupToGroup(groupIpId, ipIds[i]);
+            }
+            // check if the IP has the same license terms as the group
+            if (!LICENSE_REGISTRY.hasIpAttachedLicenseTerms(ipIds[i], groupLicenseTemplate, groupLicenseTermsId)) {
+                revert Errors.GroupingModule__IpHasNoGroupLicenseTerms(
+                    ipIds[i],
+                    groupLicenseTemplate,
+                    groupLicenseTermsId
+                );
+            }
+            pool.addIp(groupIpId, ipIds[i]);
+        }
+
+        emit AddedIpToGroup(groupIpId, ipIds);
+    }
+
+    /// @notice Removes IP from group.
+    /// the function must be called by the Group IP owner or an authorized operator.
+    /// @param groupIpId The address of the group IP.
+    /// @param ipIds The IP IDs.
+    function removeIp(address groupIpId, address[] calldata ipIds) external whenNotPaused verifyPermission(groupIpId) {
+        _checkIfGroupMembersLocked(groupIpId);
+        // remove ip from group
+        GROUP_IP_ASSET_REGISTRY.removeGroupMember(groupIpId, ipIds);
+        for (uint256 i = 0; i < ipIds.length; i++) {
+            IGroupRewardPool(GROUP_IP_ASSET_REGISTRY.getGroupRewardPool(groupIpId)).removeIp(groupIpId, ipIds[i]);
+        }
+        emit RemovedIpFromGroup(groupIpId, ipIds);
+    }
+
+    /// @notice Claims reward.
+    /// @param groupId The address of the group.
+    /// @param token The address of the token.
+    /// @param ipIds The IP IDs.
+    function claimReward(address groupId, address token, address[] calldata ipIds) external whenNotPaused {
+        IGroupRewardPool pool = IGroupRewardPool(GROUP_IP_ASSET_REGISTRY.getGroupRewardPool(groupId));
+        // claim reward from group IPA's RoyaltyVault to group pool
+        pool.collectRoyalties(groupId, token);
+        // trigger group pool to distribute rewards to group members vault
+        uint256[] memory rewards = pool.distributeRewards(groupId, token, ipIds);
+        for (uint256 i = 0; i < ipIds.length; i++) {
```

Comment:
Do we need a for loop to emit an event - could we emit rewards and ipIds instead?

Reply:
send event with array.